core/src/main/scala/kafka/admin/TopicCommand.scala

```diff
@@ -271,6 +275,49 @@ object TopicCommand extends Logging {
     ret.toMap
   }
 
+  private[admin] class PartitionsDescription(topic: String, partitions: List[(Int, Seq[Int])], liveBrokers: Set[Int], topicAndPartitionToLeaderAndIsr: Map[(String, Int), (Option[Int], Seq[Int])]) {
+
+    def describeAndPrint(): String = {
+      val sb = new StringBuilder
+      for ((partitionId, assignedReplicas) <- partitions) {
+        val (leader, inSyncReplicas) = topicAndPartitionToLeaderAndIsr.get(topic, partitionId).get
+        if (shouldPrint(assignedReplicas, leader, inSyncReplicas)) {
+          sb.append(printIt(topic, partitionId, assignedReplicas, leader, inSyncReplicas))
+        }
+      }
+      sb.toString()
+    }
+
+    def shouldPrint(assignedReplicas: scala.Seq[Int], leader: Option[Int], inSyncReplicas: scala.Seq[Int]): Boolean = true
+
+    def printIt(topic: String, partitionId: Int, assignedReplicas: scala.Seq[Int], leader: Option[Int], inSyncReplicas: scala.Seq[Int]): String = {
+      val sb = new StringBuilder
+      sb.append("\tTopic: " + topic)
+      sb.append("\tPartition: " + partitionId)
+      sb.append("\tLeader: " + (if (leader.isDefined) leader.get else "none"))
+      sb.append("\tReplicas: " + assignedReplicas.mkString(","))
+      sb.append("\tIsr: " + inSyncReplicas.mkString(","))
+
+      val description = sb.toString()
+      println(description)
+      description
+    }
+  }
+
+  private[admin] class UnderReplicatedPartitionsDescription(topic: String, partitions: List[(Int, Seq[Int])], liveBrokers: Set[Int], topicAndPartitionToLeaderAndIsr: Map[(String, Int), (Option[Int], Seq[Int])])
```

Comment:
The class hierarchy brings the following.
1. Unit tests to test that special reports conditions are valid. For instance, for a given topic under-replicated or unavailable partitions are detected correctly.
2. As these cli tools are the only way, as of now, to discover these info. Ops folks may have automation scripts around these tools that parse tools' output to get various info. Having checks to know when the output is changing will avoid unintentional changes becoming a pain for such ops guys. I personally do not know of any ops guys doing this, but this is what I have heard from multiple people.
3. Adding another special report will be clean, for instance Kafka-3247 won't have to add to the already complicated if-else logic.

Do you think above reasons justify having classes instead of a complicated if-else conditions?